codecparsers/h264Parser.cpp

```diff
@@ -349,12 +386,16 @@ bool Parser::parseSps(SharedPtr<SPS>& sps, const NalUnit* nalu)
         if (sps->seq_scaling_matrix_present_flag) {
             uint32_t loop = (sps->chroma_format_idc != 3) ? 8 : 12;
             for (uint32_t i = 0; i < loop; i++) {
-                sps->seq_scaling_list_present_flag[i] = nr.read(1);
+                READ(sps->seq_scaling_list_present_flag[i]);
                 if (sps->seq_scaling_list_present_flag[i]) {
-                    if (i < 6)
-                        scalingList(nr, sps->scaling_lists_4x4[i], 16, i);
-                    else
-                        scalingList(nr, sps->scaling_lists_8x8[i - 6], 64, i);
+                    if (i < 6) {
```

Comment:
We can use other patch to fix this, this patch only focus on check read beyond boundary things.